### ISSUE TITLE
Fix API handling of multi-column UNIQUE and PK columns

### ISSUE THREAD

**Issue by dmos62** [centerofci/mathesar#2245]
See https://github.com/centerofci/mathesar/issues/2231#issuecomment-1380309139 for some insight.

**Comment by kgodey** [comment]
Isn't this the same issue as #2231? Why do we need two separate issues?

**Comment by dmos62** [comment]
#2231 is about the frontend bug and this is about the backend bug. The underlying UX problem is singular, but its solution involves solving two different problems by two teams. That's why I created a second issue. Basically I've been been making issues for work items, not "UX problems".

If you see problems with this, let's dicuss it. I like this, because my tasks then map to issues conveniently and for example I can just have "Fixes #XXXX" on top of each PR and GH will autoclose the parent issue. Otherwise, I have to do "Related #XXXX", because I'm only solving part of the problem, and then I have to remember/track remaining work in #XXXX in some other way, which is a drain on my capacity to task manage.

**Comment by kgodey** [comment]
@dmos62 I think it's good to have separate issues from backend and frontend. It wasn't apparent to me from the descriptions. I think it would be helpful to update the description of #2231 to focus it on the frontend and also mark it as blocked by this issue.

**Comment by dmos62** [comment]
@kgodey, done.

**Comment by Hatilar420** [comment]
Hi @dmos62 , can i work on this?

**Comment by dmos62** [comment]
@Hatilar420 sure. This is a less clear cut task though. Be proactive about making sure your understanding of the task is in sync with ours.

**Comment by Hatilar420** [comment]
Hi @dmos62, are there any pre-existing tables in the codebase or a URL? That might help me recreate this issue quickly.

**Comment by dmos62** [comment]
@Hatilar420 I don't think so.

**Comment by Hatilar420** [comment]
Hi @dmos62, I am able to render tables with multi-column PK constrained. Which is opposed to the observed behaviour in #2231 . However, when I try to patch or delete any cell, the server throws a 500 internal error.

- Attached is SS for multiple constrained PK rendering.
![image](https://user-images.githubusercontent.com/29446639/224211960-44475b97-ba9d-448e-b75c-90f8a928daf0.png)



Is this the correct behaviour I am observing ?

**Comment by dmos62** [comment]
As I see it there are at least two tasks here:
- 500s should not happen, 
- Backend should accurately describe the primary and unique constraints (including when they are multi-column) associated with a table to the frontend.

**Comment by Hatilar420** [comment]
Hi @dmos62, It might take some time, which can be more than a week, to open a PR for this issue as I am busy with some college work and some work from my internship. Is it okay with the contributor rules and with you?

**Comment by Hatilar420** [comment]
Yes, sounds good. Thank you @dmos62  :) .

**Comment by Hatilar420** [comment]
Hi @dmos62, I am sorry for this late reply, I was busy with the GSOC proposal and college Assignments.
An update so far is that. I was able to fix this issue where the first column. (ie the id column is unique). However, in the cases where the `Id` column is not Unique, since we are working with composite Pkeys and the `Id` column can have the same value in some records, I was able to delete and patch(with the data which is meant for another record) multiple records having the same `id` values. 
Do we want this behaviour?
